### PR TITLE
[kotlin-spring] Fix cookie parameter code generation in API

### LINKED ISSUE #7406: [BUG] [Kotlin | Spring] Cookie in parameter is not correctly generated
This has already been fixed for Java in #5386
Sadly this issue is also in the Kotlin Spring Boot generator.

### PR DESCRIPTION
**Pull request by oscarr-reyes**
Fixes #7406

When generating API interfaces or class controllers from a 3.0 spec file that contains cookie parameter in the request, the output code doesn't contain the endpoint parameter for resolving the cookie value that is in the spec, the template files don't contain the section for cookie and since it's part of `allParams` section an empty space is generated with a comma, making the generated code to be erroneous.

This PR adds the required template code for generating cookie parameter values from the spec file.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by oscarr-reyes** [comment]
@jimschubert @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m @stefankoppier

File: bin/configs/kotlin-spring-boot-request-cookie.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+generatorName: kotlin-spring
+outputDir: samples/server/petstore/kotlin-springboot-request-cookie
```
Comment (wing328) [inline bin/configs/kotlin-spring-boot-request-cookie.yaml:2]: i ran `gradle build` on this folder but got lots of errors.

did you get errors when trying to build the project locally?

Comment (oscarr-reyes) [reply bin/configs/kotlin-spring-boot-request-cookie.yaml:2]: I tried to run the project and my local gradle isn't even able to create the tasks.. i tried the same with the other samples but got the same result so i wasn't sure if that was expected, maybe it's a gradle version issue. I created this PR and wait to see if that was an issue or just something to ignore.

Comment (wing328) [reply bin/configs/kotlin-spring-boot-request-cookie.yaml:2]: I ran `mvn clean package` in this output folder but got some errors:
```
\org\openapitools\api\AnotherFakeApi.kt: (42, 19) Unresolved reference: another
[ERROR] C:\Users\wing3\Code\openapi-generator\samples\server\petstore\kotlin-springboot-request-cookie\src\main\kotlin\org\openapitools\api\FakeApi.kt: (422, 126) Parameter name expected
[ERROR] C:\Users\wing3\Code\openapi-generator\samples\server\petstore\kotlin-springboot-request-cookie\src\main\kotlin\org\openapitools\api\FakeApi.kt: (422, 127) Expecting comma or ')'
[ERROR] C:\Users\wing3\Code\openapi-generator\samples\server\petstore\kotlin-springboot-request-cookie\src\main\kotlin\org\openapitools\api\FakeApi.kt: (422, 127) Expecting ')'

```

can you please take a look?

`mvn clena package` works fine with the master using the config/spec you provided in this PR

Comment (oscarr-reyes) [reply bin/configs/kotlin-spring-boot-request-cookie.yaml:2]: I've got different errors but i figured what was causing them, the spec file that's being used is a copy from a `fake-petstore` file for version 3 which has some cases that are not properly handled in `kotlin-spring` generator unless the configuration for the generator is not correct for those cases.

I've removed all the fake cases except the cookie one in the spec and now the sample doesn't throw any error from my end, feel free to run the tasks again and let me know if there are any other issues.

**Comment by wing328** [comment]
lgtm. thanks for the PR

**Comment by wing328** [comment]
FYI. samples updated via 8dae7026b9a

**Pull request merged**